puppet-squid/src/main/java/com/iadams/sonarqube/puppet/api/PuppetGrammar.java

```diff
@@ -162,7 +162,10 @@ public static void grammar(LexerfulGrammarBuilder b) {
                 ARGUMENT_EXPRESSION_LIST,
                 b.optional(RPAREN));
 
-        b.rule(ARGUMENT_EXPRESSION_LIST).is(EXPRESSION, b.zeroOrMore(COMMA, EXPRESSION), b.optional(COMMA));
+        b.rule(ARGUMENT_EXPRESSION_LIST).is(
+                b.firstOf(EXPRESSION, VARIABLE, QUALIFIED_IDENTIFIER, IDENTIFIER),
+                b.zeroOrMore(COMMA, EXPRESSION),
```

Comment:
@iwarapter: The grammar is getting quite complex... There's something wrong here. Instead of ```EXPRESSION```, there should be ```EXPRESSION, VARIABLE, QUALIFIED_IDENTIFIER, IDENTIFIER```. It parses for now because all  the arguments after the first one are considered as function calls even if they're not.